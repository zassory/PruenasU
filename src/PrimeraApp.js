//import React , { Fragment } from 'react';
import React from 'react';
import propTypes from 'prop-types';

const PrimeraApp = ( { saludo, subtitulo } ) => {
        
    return (
        <>
            <h1>{ saludo }</h1>
            {/* <pre> { JSON.stringify( saludo , null , 3) } </pre> */}
            <p>{ subtitulo }</p>
        </>
    );                    
}

PrimeraApp.propTypes = {
    saludo: propTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;