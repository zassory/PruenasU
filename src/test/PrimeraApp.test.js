import React from 'react';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

//import '@testing-library/jest-dom';
//import { render } from '@testing-library/react';

describe('Pruebas en <PrimeraApp />' , ()=> {

    // test('Debe mostrar el mensaje "Hola, soy Goku"', () => { 
    //     const saludo = 'Hola, soy Goku';
    //     const { getByText } = render( <PrimeraApp saludo={ saludo }/> )
    //     expect( getByText(saludo) ).toBeInTheDocument();
    // });

    test('Debe mostrar <PrimeraApp /> correctamente', () => { 

        const saludo = 'Hola, soy Goku';
        const wrapper = shallow(<PrimeraApp saludo = { saludo } />);

        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de mostrar el sutitulo enviado por props', () => { 

        const saludo = 'Hola, soy Goku';
        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow(
            <PrimeraApp 
            saludo = { saludo }
            subtitulo = { subtitulo }
            />
        );

        const textoParrafo = wrapper.find('p').text();
        expect( textoParrafo ).toBe(subtitulo);

    });


})