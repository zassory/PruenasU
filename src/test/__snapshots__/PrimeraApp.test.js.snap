// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraAPp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
